README.md

```diff
@@ -1,9 +1,39 @@
 # vIBC Core Smart Contracts
 
-This project includes
+This project includes the core smart contracts for the vIBC protocol, and a few demo contracts that simulate testing and serve as a template for integrating dapp devs.
 
-- vIBC Core Smart Contracts (CoreSC)
-- a few demo contracts that simulate dev users protocol contracts, eg. Mars, etc.
+![](./diagrams/vibcContractsOverview.jpg)
+
+
+## Repo Structure
+
+All contracts internal to this project are in the `contracts`. This directory contains the following subdirectories:
+- `core/`: The contracts that are core to the vIBC protocol. These are the contracts that will be deployed and used in production. 
+- `interfaces/`: Interfaces for core and testing contracts 
+- `libs/`: Libraries used by the core vIbc protocol.
+- The `utils/`, `base/`, and `example/` directories all contain contracts that are not core to the vIBC protocol. These contracts are used only for testing or as templates for dapp contracts that will integrate with the vIBC protocol. 
+
+
+# Core Contracts
+## Dispatcher
+The Dispatcher contract routes packets to dapps by mapping channels to dapp addresses. The dispatcher also calls packet and channel callbacks on sending/receiving dapps. This contract uses a UUPS proxy compliant with ERC1967 for upgrades. The proxy is intended to be managed by a single owner address, which will be replaced by a multisig in production. 
+
+Since the optimistic light client contract is used to prove that events happened on the peptide chain, all methods in the dispatcher are permissionless, aside from the methods to set the connection to client mapping, upgrading the implementation, and setting the dispatcher's port prefix. 
```

Comment:
```suggestion
Since the optimistic light client contract is used to prove that events happened on the Polymer chain, all methods in the dispatcher are permissionless, aside from the methods to set the connection to client mapping, upgrading the implementation, and setting the dispatcher's port prefix. 
```